py-polars/polars/internals/series.py

```diff
@@ -341,100 +335,50 @@ def __ge__(self, other: Any) -> "Series":
-                return Series._from_pyseries(self._s.div(other._s))
-
-            other = maybe_cast(other, self.dtype)
-            f = get_ffi_func("div_<>", self.dtype, self._s)
-            if f is None:
-                return NotImplemented
-            return wrap_s(f(other))
+            return self._arithmetic(other, "div", "div_<>")
 
         return self.cast(Float64) / other
 
     def __floordiv__(self, other: Any) -> "Series":
-        if isinstance(other, Series):
-            if self.is_float():
-                return Series._from_pyseries(self._s.div(other._s)).floor()
-            return Series._from_pyseries(self._s.div(other._s))
-
-        other = maybe_cast(other, self.dtype)
-        f = get_ffi_func("div_<>", self.dtype, self._s)
-        if f is None:
-            return NotImplemented
+        result = self._arithmetic(other, "div", "div_<>")
         if self.is_float():
-            return wrap_s(f(other)).floor()
-        return wrap_s(f(other))
+            result = result.floor()
+        return result
 
     def __mul__(self, other: Any) -> "Series":
-        if isinstance(other, Series):
-            return Series._from_pyseries(self._s.mul(other._s))
-        other = maybe_cast(other, self.dtype)
-        f = get_ffi_func("mul_<>", self.dtype, self._s)
-        if f is None:
-            return NotImplemented
-        return wrap_s(f(other))
+        return self._arithmetic(other, "mul", "mul_<>")
 
     def __mod__(self, other: Any) -> "Series":
-        if isinstance(other, Series):
-            return Series._from_pyseries(self._s.rem(other._s))
-        other = maybe_cast(other, self.dtype)
-        f = get_ffi_func("rem_<>", self.dtype, self._s)
-        if f is None:
-            return NotImplemented
-        return wrap_s(f(other))
+        return self._arithmetic(other, "rem", "rem_<>")
 
     def __rmod__(self, other: Any) -> "Series":
-        if isinstance(other, Series):
-            return Series._from_pyseries(other._s.rem(self._s))
-        other = maybe_cast(other, self.dtype)
-        other = match_dtype(other, self.dtype)
-        f = get_ffi_func("rem_<>_rhs", self.dtype, self._s)
-        if f is None:
-            return NotImplemented
-        return wrap_s(f(other))
+        return self._arithmetic(other, "rem", "rem_<>_rhs")
 
     def __radd__(self, other: Any) -> "Series":
-        if isinstance(other, Series):
-            return Series._from_pyseries(self._s.add(other._s))
-        other = maybe_cast(other, self.dtype)
-        other = match_dtype(other, self.dtype)
-        f = get_ffi_func("add_<>_rhs", self.dtype, self._s)
-        if f is None:
-            return NotImplemented
-        return wrap_s(f(other))
+        return self._arithmetic(other, "add", "add_<>_rhs")
 
     def __rsub__(self, other: Any) -> "Series":
-        if isinstance(other, Series):
-            return Series._from_pyseries(other._s.sub(self._s))
-        other = match_dtype(other, self.dtype)
```

Comment:
This is an example of just `match_dtype`. Note that `maybe_cast` does the same casting, except it will not force strings and bools to integers.

Reply:
I cannot recall. Why did we cast strings to integers in the first place? 

Oh maybe to allow for `[1, 2, 3] - true`? Which is edgy IMO.

Reply:
Yes, I think any behaviour that may change due this PR is edgy to start with. If we see a valid use case, we can add that, and it would also automatically apply for all arithmetic methods.

Reply:
Yes, agree.